relay-server/src/utils/stask.rs

```diff
@@ -0,0 +1,197 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use crate::utils::ScheduledQueue;
+use futures::{
+    stream::{FusedStream, FuturesUnordered},
+    Stream, StreamExt,
+};
+use tokio::time::Instant;
+
+/// A set of tasks/futures that can be scheduled for execution.
+#[derive(Debug)]
+#[must_use = "streams do nothing unless polled"]
+pub struct ScheduledTasks<T> {
```

Comment:
nit(naming): Should we name this `TaskRunner` or similar to emphasize that this is the component that does the work?